### PR TITLE
Added microphone demo, updated README, added requirements_gpu.txt 

### PR DESCRIPTION
**Pull request by mallorbc**
This PR adds a CLI tool that allows a user to use a microphone to interact with the models.  It also updates the readme and adds a requirements.txt for GPU usage.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,3 +4,7 @@ tqdm
 more-itertools
 transformers>=4.19.0
 ffmpeg-python==0.2.0
+click
+pyaudio
+SpeechRecognition
+pydub
```
Comment (shirayu) [inline requirements.txt:10]: These packages are only needed when running the demo and should be declared as extra in `setup.py ` and excluded from `install_requires`.
(The current ``setup.py`` make all packages in requirements.txt `install_requires`.)

**Comment by jongwook** [comment]
Thank you for creating this demo. I'm trying to keep this repo minimal, containing only the necessary code to run inference and evaluation. Would you be interested in making this a separate GitHub project and maybe post in the "Show and tell" category in the Discussions tab?

Nice point about the GPU requirements - I'm planning to add more notes in README.md about ways to install device-accelerated PyTorch versions, esp. on Windows and MacOS.

**Comment by GreenMan36** [comment]
I'm very interested in using this to make a tool for deaf people.
My hope is that this will be able to take system audio for voicechat or in-game dictation.
Possibly real-time translation as well to make it useful for everyone and not just deaf people.

**Comment by shirayu** [comment]
FYI: #2 is the discussion related to this.
I'm also working about this at https://github.com/shirayu/whisper_streaming

**Comment by mallorbc** [comment]
@jongwook I understand, that makes sense.  I will close this PR and make my own repo

**Comment by mallorbc** [comment]
I have released the code under the AGPL license here.  There is no license to use the code in the PR under MIT since it was not merged.
https://github.com/mallorbc/whisper_mic

**Pull request closed without merging**